pkg/registry/core/service/storage/rest.go

```diff
@@ -162,8 +218,44 @@ func (rs *REST) Export(ctx context.Context, name string, opts metav1.ExportOptio
 	return rs.services.Export(ctx, name, opts)
 }
 
+func (rs *REST) isExternalIPAM(ctx context.Context, service *api.Service) (bool, error) {
+	netName := service.Labels[arktosextensionsv1.NetworkLabel]
+	if netName == "" {
+		// choose the default network name if not specified
+		netName = "default"
+	}
+
+	network, err := rs.getNetwork(context.Background(), service.Tenant, netName)
+	if err != nil {
+		// for now, temporarily ignore the default network not found error
+		// todo: remove below line after tenant controller enables automatic default network creation on new tenants
+		if netName == "default" {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	net := network.(*unstructured.Unstructured)
+	netSpec := net.Object["spec"].(map[string]interface{})
+	if s, ok := netSpec["service"]; ok {
+		netSvc := s.(map[string]interface{})
+		if v, ok := netSvc["ipam"]; ok {
+			if vs, ok := v.(string); ok {
+				return vs == string(arktosextensionsv1.IPAMExternal), nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
 func (rs *REST) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
```

Comment:
Looks to me we need some logic in Update() so the system cannot assign IP to an External network.

Reply:
yes thanks for the feedback. will modify update method.